backend/src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,36 @@
+package com.onlinebookstore.repository.book;
+
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.model.SearchParameters;
+import com.onlinebookstore.repository.SpecificationBuilder;
+import com.onlinebookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(SearchParameters searchParameters) {
+        Specification<Book> specification = Specification.where(null);
+        if (searchParameters.author() != null && !searchParameters.author().isEmpty()) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("author")
+                    .getSpecification(searchParameters.author()));
+        }
+        if (searchParameters.isbn() != null && !searchParameters.isbn().isEmpty()) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("isbn")
+                    .getSpecification(searchParameters.isbn()));
+        }
+        if (searchParameters.title() != null && !searchParameters.title().isEmpty()) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("title")
+                    .getSpecification(searchParameters.title()));
+        }
```

Comment:
very similar code
try to make loop with predefined list of properties
most probably you will need ReflectionUtils library to handle such approach